floto/specs/task/task.py

```diff
@@ -23,5 +38,26 @@ def _default_id(self, name, version, input):
         else:
             requires_string = ''
 
-        hash_ = hashlib.sha1((input_string+requires_string).encode()).hexdigest()[:10]
-        return '{}:{}:{}'.format(name, version, hash_)
+        hash_ = hashlib.sha1((input_string + requires_string).encode()).hexdigest()[:10]
+        return '{}:{}:{}:{}'.format(name, version, domain, hash_)
+
+    def serializable(self):
+        """Serializable representation of self
+
+        Returns
+        -------
+        dict
+        """
+        # remove None values
+        cpy = {k: v for k, v in self.__dict__.items() if v is not None}
+
+        # add type
+        module_name = '.'.join(self.__module__.split('.')[:-1])
+        cpy['type'] = module_name + '.' + self.__class__.__name__
+
+        # in the requires list, we need nothing but the task's id_
+        requires = cpy.get('requires')
+        if requires:
+            cpy['requires'] = [{'id_': t.id_} for t in requires]
```

Comment:
@amatthies 
This does currently not work because the dicts with id_ are not getting deserialized to specs.Task objects. On the other hand the ExecutionGraph is working with Task objects in the 'requires' field internally. We have the following options:

1. requires must be of type list(str) with the task ids; This would slightly change the interface to e.g ActivityTask(...requires=[task_x.id_]...)
2. We add the type information to the dict in line 61
3. We change the ExecutionGraph such that it can cope with lists of tasks and lists of ids
4. We leave the floto api as it is, i.e. ActivityTask(...requires=[task_x]...) and transform it internally to a list of ids (and change ExecutionGraph to cope with list of ids)